src/web/js/trove/chart-lib.js

```diff
@@ -70,6 +71,28 @@
     return rgb2hex(IMAGE.colorString(checkColor(v)));
   }
 
+  function convertPointer(p) {
+    return {v: toFixnum(p.dict.value) , f: p.dict.label}
+  }
+
+  // Converts a list with func:
+  function convertListWith(func, l) {
```

Comment:
I think this is unsafe. JS Stack might overflow. I'm also not sure if using `val.dict.first` as a terminating condition is actually OK. In any case, I think there already exists a similar but safe function in core Pyret already, so you should just use it. Another possibility is to convert a list to a (raw-)array from Pyret side.

Reply:
Got it! I’ll convert the lists into raw-arrays on Pyret's side in the next commit.